clippy2021.md

```diff
@@ -0,0 +1,178 @@
+classifying lints. This will also involve internal changes, and will be
+addressed later in this document again.
+
+[#5537](https://github.com/rust-lang/rust-clippy/issues/5537)
+[#6366](https://github.com/rust-lang/rust-clippy/issues/6366)
+
+#### `lints.toml` configuration
+
+This is something that comes up every now or then. A configuration file, where
+lint levels can be defined and reused across projects. Discussions about this
+often lead to nothing specific/actionable or to "we need an RFC for this". And
+this is exactly what we need to do. Get together with the cargo team and write
+an RFC and implement such a configuration file somehow and somewhere.
+
+[#3164](https://github.com/rust-lang/rust-clippy/issues/3164)
+[cargo#5034](https://github.com/rust-lang/cargo/issues/5034)
+[IRLO](https://internals.rust-lang.org/t/proposal-cargo-lint-configuration/9135/8)
+
+
+#### FP rate
+
+In the worst case, new lints are only available in nightly for 2 weeks, before
+hitting beta and ultimately stable. This and the fact that fewer people use
+nightly Rust nowadays makes it more probable that a lint with many FPs hits
+stable. This leads to annoyed users, that will disable these new lints in the
+best case and to more annoyed users, that will stop using Clippy in the worst.
+We should think of a process to prevent this from happening. This will also be
+addressed later in this document.
+
+[#6429](https://github.com/rust-lang/rust-clippy/issues/6429)
+
+#### Clippy book
+
+We should consider writing a Clippy book, like other projects have. This will
+then be the central place where everything you need to know to use Clippy is
+collected in a structured way.
+
+[#6011](https://github.com/rust-lang/rust-clippy/issues/6011)
+
+### Reliability
+
+We should make sure to make Clippy more reliable and stable for our users
+
+#### FP rate
+
+In the worst case, new lints are only available in nightly for 2 weeks, before
+hitting beta and ultimately stable. This and the fact that fewer people use
+nightly Rust nowadays makes it more more probably that a lint with many FPs hits
+stable. This leads to annoyed users, that will disable these new lints in the
+best case and to more annoyed users, that will stop using Clippy in the worst.
+We should think of a process to prevent this from happening. This will also be
+addressed later in this issue.
+
+[#6429](https://github.com/rust-lang/rust-clippy/issues/6429)
+
+## Internal
+
+(The end of) 2020 has shown, that Clippy has to think about the available
+resources, especially regarding management and maintenance of the project. This
+section address issues affecting team members and contributors.
+
+### Management
+
+In 2020 we got over 1000 open issues with now between 25-35 open PRs. This is
+mostly because of the lack of resources in the team. According to the planned
+roadmap of the Rust project, we should improve our processes and look for more
+eyes and hands to help reviewing and triaging PRs and issues.
```

Comment:
I would love to chat if you want about some of the attempts I've made historically on rust-lang/rust to deal with issues (though largely all of them have not been super successful, as our 6,000+ issues may show...). I know @jonas-schievink has been doing issue triage for the past year or two as well, too, and maybe something like the prioritization group process may be helpful too.

In general, this is an area where as Rust grows, we're going to want to work out better processes for dealing with issue influx. I don't know yet what those are, but I imagine that we're going to end up with the same problems (though maybe not the same solutions!) on many of our repositories :)

Reply:
I see, keeping this issue team-internal worked out well. You're just everywhere 😄 

But I would love to figure out things we can do across projects to improve triaging. For Clippy specifically, I wrote a small program before, which creates a list of issues sorted by lint (which never got anywhere). Another idea I had for PRs is to write a dashboard similar to the bors queue, which tracks the last activity on PRs, with information about the reviewer/author/waiting-on-{review,author} state.

Reply:
On triage: it's worth wondering if we can loop in the existing triage team to help. If we wrote a TRIAGING doc that lets others triage things here that would also be cool.

re: team-internal: sorry, i posted it in #core because i thought it was a really well designed roadmap, as a "this is cool", and particularly wanted Mark to have a look because he's been doing the main roadmap 😄